modulus/models/graphcast/graph_cast_net.py

```diff
@@ -43,6 +43,53 @@
 from .graph_cast_processor import GraphCastProcessor
 
 
+def get_lat_lon_partition_separators(partition_size: int):
+    """Utility Function to get separation intervals for lat-lon
+    grid for partition_sizes of interest.
+
+    Parameters
+    ----------
+    partition_size : int
+        size of graph partition, currently supported sizes: {1, 2, 4, 8, 16}
+    """
+
+    if partition_size == 1:
```

Comment:
I wish we could make this to work with any `partition_size`.

Reply:
I updated the logic to make it more general. Not it should support any `partition_size` but defaults back to just splitting the domain into `partition_size` along the longitude values. 
We should have somewhat balanced distribution schemes for `partition_size in {2, 4, 6, 8, 9, 12, 16, 20, 24}` which likely will cover most of the relevant sizes for now anyways.

Reply:
Can we also have a default where we just take the two closest factors (closest to `sqrt(partition_size)`)?

Reply:
sure, this is actually the better solution for the "generic" case anyways and should even cover all the special cases

Reply:
done